### PR TITLE
fix: fixing lost sync of off products on ovh3

### PR DESCRIPTION
**Pull request by alexgarel**
### What

Problem of sync on ovh3, with a blocking situation.

### CODE REVIEW

File: docs/reports/2023-06-08-stalled-zfs-sync-off-products.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Restart staging, on docker-staging: `cd /home/off/off-net; sudo -u off docker-compose start`
+
+
+### Relaunch sync manually
+
+To better follow what's happening I decided to manually rollback to the common snapshot: `20230501-0000`
+
+on ovh3:
+```bash
+zfs rollback -r rpool/off/products@20230501-0000
```
Comment (stephanegigandet) [inline docs/reports/2023-06-08-stalled-zfs-sync-off-products.md:52]: sorry if this is a stupid question, but I don't understand:
just before, you had cloned that snapshot on ovh3: zfs clone rpool/off/products@20230501-0000 rpool/off/clones/products
so why was rollback to the same snapshot needed?

File: docs/reports/2023-06-08-stalled-zfs-sync-off-products.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+```
+I didn't measure but it took something between 20 and 40 minutes.
+
+on off1:
+```bash
+zfs send -i rpool/off/products@20230501-0000 rpool/off/products@20230608-1130 | ssh ovh3.openfoodfacts.org zfs recv rpool/off/products -F
```
Comment (stephanegigandet) [inline docs/reports/2023-06-08-stalled-zfs-sync-off-products.md:58]: and here why do you send the rpool/off/products@20230501-0000 snapshot? isn't it already on ovh3?

File: scripts/off1/sto-products-sync.sh [hunk 3]
```diff
@@ -47,15 +51,15 @@ do
...
+  OLD=$(zfs list -t snap -o name | grep "$DATASET@$(date -d '2 days ago' +%Y%m%d)" | tail -n +2)
   for OLD_SNAP in $OLD
   do
     zfs destroy $OLD_SNAP
   done
 
-  # suppression des snapshots du mois dernier sauf le premier
-  OLD=$(zfs list -t snap -o name | grep "$DATASET@$(date -d 'last month' +%Y%m)" | tail -n +2)
+  # suppression des snapshots de mois -2 sauf le premier
+  OLD=$(zfs list -t snap -o name | grep "$DATASET@$(date -d '2 month days' +%Y%m)" | tail -n +2)
```
Comment (stephanegigandet) [inline scripts/off1/sto-products-sync.sh:62]:
```suggestion
  OLD=$(zfs list -t snap -o name | grep "$DATASET@$(date -d '2 month ago' +%Y%m)" | tail -n +2)
```

File: docs/reports/2023-06-08-stalled-zfs-sync-off-products.md (hunk 1 above)
Comment (alexgarel) [reply docs/reports/2023-06-08-stalled-zfs-sync-off-products.md:52]: The clone creates a new dataset based on a snapshot.

Meanwhile the main dataset (rpool/off/products) is still using the newest snapshot.

Think of it like in git, if you derive a new branch from an old commit (the new branch is the clone). 
The "main" branch is still on a different commit (the newest one).

By doing the rollback, it's like changing "main" branch to point to an old commit, and force pushing it !

File: scripts/off1/sto-products-sync.sh (hunk 3 above)
Comment (alexgarel) [reply scripts/off1/sto-products-sync.sh]: Thanks for spotting it !

**Pull request merged**